core/commands/config.go

```diff
@@ -219,22 +250,20 @@ func getConfig(r repo.Repo, key string) (*ConfigField, error) {
 }
 
 func setConfig(r repo.Repo, key string, value interface{}) (*ConfigField, error) {
-	err := r.SetConfigKey(key, value)
+	keyF, err := getConfig(r, "Identity.PrivKey")
 	if err != nil {
-		return nil, fmt.Errorf("Failed to set config value: %s (maybe use --json?)", err)
+		return nil, errors.New("failed to get PrivKey")
 	}
-	return getConfig(r, key)
-}
-
-func showConfig(filename string) (io.Reader, error) {
-	// TODO maybe we should omit privkey so we don't accidentally leak it?
-
-	data, err := ioutil.ReadFile(filename)
+	privkey := keyF.Value
+	err = r.SetConfigKey(key, value)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to set config value: %s (maybe use --json?)", err)
 	}
-
-	return bytes.NewReader(data), nil
+	err = r.SetConfigKey("Identity.PrivKey", privkey)
```

Comment:
I think we should use this opportunity to remove the private key from the config file altogether. `Identity.PeerID` can still remain as a readonly attribute for compat. The preferred way of getting one's own PeerID is `:5001/api/v0/id` and we should state that somewhere.